sriovnet_switchdev.go

```diff
@@ -183,6 +183,33 @@ func findNetdevWithPortNameCriteria(criteria func(string) bool) (string, error)
 	return "", fmt.Errorf("no representor matched criteria")
 }
 
+// SetVfRepresentorSmartNICMacAddress sets the given MAC address for a host VF identified by pfID and vfIndex
+func SetVfRepresentorSmartNICMacAddress(pfID, vfIndex string, mac net.HardwareAddr) error {
+	// pfID should be 0 or 1
+	if pfID != "0" && pfID != "1" {
+		return fmt.Errorf("unexpected pfID(%s). It should be 0 or 1", pfID)
+	}
+
+	// vfIndex should be an unsinged integer provided as a decimal number
+	if _, err := strconv.ParseUint(vfIndex, 10, 32); err != nil {
+		return fmt.Errorf("unexpected vfIndex(%s). It should be an unsigned decimal number", vfIndex)
+	}
+
+	uplinkPortName := fmt.Sprintf("p%s", pfID)
+	vfRepName := fmt.Sprintf("vf%s", vfIndex)
+	sysfsVfRepMacFile := filepath.Join(NetSysDir, uplinkPortName, "smart_nic", vfRepName, "mac")
```

Comment:
uplink netdev is not necessarily named as the uplink port name.

you should search for a netdev that its phys_port_name matches `uplinkPortName`. see Lines 345-350 as they do something similar.